### PR TITLE
[docs] Make some corrections and additions to UnderscoredAttributes.md

### PR DESCRIPTION
**Pull request by varungandhi-apple**
It seems like `@_assemblyVision` was added recently, so I added a description for it and did some other minor cleanup.

### CODE REVIEW

File: docs/ReferenceGuides/UnderscoredAttributes.md [hunk 1]
```diff
@@ -39,6 +39,11 @@ Most notably, default argument expressions are implicitly
 `@_alwaysEmitIntoClient`, which means that adding a default argument to a
 function which did not have one previously does not break ABI.
 
+## `@_assemblyVision`
+
+Forces emission of optimization remarks (such as information about retains
```
Comment (gottesmm) [inline docs/ReferenceGuides/UnderscoredAttributes.md:44]: It isn't optimization remarks. I would say:

Forces emission of Assembly Vision Remarks that show where various runtime calls and performance impacting hazards are in the code at source level after optimization.

Comment (varungandhi-apple) [reply docs/ReferenceGuides/UnderscoredAttributes.md:44]: Hmm, I based my description based on the test cases. Here is one of them:

```swift
@_semantics("optremark")
public func mix4() -> (Klass, Klass) {
    let x = getGlobal()
    return (x, Klass()) // expected-remark {{Pure call. Always profitable to inline "main.Klass.__allocating_init()"}}
                        // expected-remark @-1 {{heap allocated ref of type 'Klass'}}
}

public func mix5() -> (Klass, Klass) {
    let x = getGlobal()
    return (x, Klass())
}

@_assemblyVision
public func mix4a() -> (Klass, Klass) {
    let x = getGlobal()
    return (x, Klass()) // expected-remark {{Pure call. Always profitable to inline "main.Klass.__allocating_init()"}}
                        // expected-remark @-1 {{heap allocated ref of type 'Klass'}}
}
```

That makes it seem like they're doing the same thing?

Comment (gottesmm) [reply docs/ReferenceGuides/UnderscoredAttributes.md:44]: @varungandhi-apple it builds on top of opt-remarks. I am going to actually be adding in a filtering functionality. I just haven't had time to do it.

Comment (gottesmm) [reply docs/ReferenceGuides/UnderscoredAttributes.md:44]: The reason why I put it in like that today is sometimes it helps you to reason about why a retain/release is in a specific function or not by reasoning about inlining. With time I am going to be able to do a better job and not need that.

Comment (varungandhi-apple) [reply docs/ReferenceGuides/UnderscoredAttributes.md:44]: Adjusted description based on feedback.

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

**Pull request merged**